### PR TITLE
docs(firebase_messaging): simplify notifications example

### PR DESCRIPTION
**Pull request by Ascenio**
## Description

This PR modifies the documentation to use a better practice which is to centralize message handling in one place.

## Related Issues

None.

## Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/messaging/notifications.mdx [hunk 1]
```diff
@@ -114,19 +114,19 @@ class _Application extends State<Application> {
...
-        Navigator.pushNamed(context, '/chat',
-          arguments: ChatArguments(message));
-      }
-    });
+    FirebaseMessaging.onMessageOpenedApp.listen(_handleMessage);
+  }
+  
+  Future<void> _handleMessage(RemoteMessage message) {
+    if (message.data['type'] == 'chat') {
+      Navigator.pushNamed(context, '/chat', arguments: ChatArguments(message));
```
Comment (russellwheatley) [inline docs/messaging/notifications.mdx:128]:
```suggestion
      Navigator.pushNamed(context, '/chat', 
        arguments: ChatArguments(message),
      );
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: docs/messaging/notifications.mdx (hunk 1 above)
Comment (Ascenio) [reply docs/messaging/notifications.mdx:128]: Right. I didn't use an IDE so wasn't obvious to me that we were breaking the column limit. Going to fix it.

**Comment by russellwheatley** [comment]
Thanks, @Ascenio 👍

File: docs/messaging/notifications.mdx [hunk 2]
```diff
@@ -114,19 +114,21 @@ class _Application extends State<Application> {
 
     // If the message also contains a data property with a "type" of "chat",
     // navigate to a chat screen
-    if (initialMessage != null && initialMessage.data['type'] == 'chat') {
-      Navigator.pushNamed(context, '/chat',
-          arguments: ChatArguments(initialMessage));
+    if (initialMessage != null) {
+      await _handleMessage(initialMessage);
```
Comment (russellwheatley) [inline docs/messaging/notifications.mdx:118]: Sorry, need to also remove `await` as it isn't a Future

Comment (russellwheatley) [reply docs/messaging/notifications.mdx:118]:
```suggestion
      _handleMessage(initialMessage);
```

File: docs/messaging/notifications.mdx [hunk 3]
```diff
@@ -114,19 +114,21 @@ class _Application extends State<Application> {
...
-    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
-      if (message.data['type'] == 'chat') {
-        Navigator.pushNamed(context, '/chat',
-          arguments: ChatArguments(message));
-      }
-    });
+    FirebaseMessaging.onMessageOpenedApp.listen(_handleMessage);
+  }
+  
+  Future<void> _handleMessage(RemoteMessage message) {
```
Comment (russellwheatley) [inline docs/messaging/notifications.mdx:126]:
```suggestion
  void _handleMessage(RemoteMessage message) {
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Sorry, @Ascenio. A couple more changes I didn't pick up the first time. Thanks.

**Pull request merged**